### PR TITLE
Vendor `gardener@1.80` and Upgrade `k8s.io/* to v0.28`, `sigs.k8s.io/controller-runtime to v0.16`

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area open-source
/kind enhancement

**What this PR does / why we need it**:
- Vendor gardener v1.80.0
- Upgrade k8s.io/* to v0.28
- Upgrade sigs.k8s.io/controller-runtime to v0.16

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
The following dependency is updated:
- github.com/gardener/gardener: v1.75.0-> v1.80.1
- k8s.io/* : v0.26.3 -> v0.28.2
- sigs.k8s.io/controller-runtime: v0.14.6-> v0.16.2
```

### CODE REVIEW

File: hack/tools.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ import (
 	_ "github.com/gardener/gardener/hack/tools/report-collector"
 
 	_ "github.com/ahmetb/gen-crd-api-reference-docs"
+	_ "go.uber.org/mock/gomock"
```
Comment (vpnachev) [inline hack/tools.go:18]: What is the reason this package to be imported here?

Comment (acumino) [reply hack/tools.go:18]: All tools that are required need to be pulled forcefully ref- 
https://github.com/gardener/gardener/blob/2437fcf30f80ffecedb4b823551d81c0c91fc69c/hack/tools.go#L18-L33

Comment (vpnachev) [reply hack/tools.go:18]: But `github.com/golang/mock` was never a dependency of the `hack` module, why does the replacement `go.uber.org/mock/gomock` is now made a dependency?

Comment (acumino) [reply hack/tools.go:18]: Yes, it was not, `go.uber.org/mock/gomock` can also be removed as `go.uber.org/mock/gomock` is pulled anyways by other packages.

Comment (acumino) [reply hack/tools.go]: Removed it in 9e408b0

**Review by vpnachev — APPROVED** [review]
/lgtm

**Review by dimityrmirchev — CHANGES_REQUESTED** [review]
Thank you! Can you please also update [this](https://github.com/gardener/gardener-extension-shoot-oidc-service/blob/d9cc5fd9989b1f32ecee11f5b1af4cd5630094a3/hack/test-e2e-provider-local.sh#L18) with the commit hash of g/g 1.80.1?

**Review by dimityrmirchev — APPROVED** [review]
/lgtm

**Pull request merged**